src/components/node-list/node-list-items.js

```diff
@@ -159,13 +165,63 @@ export const getFilteredModularPipelineItems = createSelector(
         faded: false,
         visible: true,
         disabled: false,
-        unset: false,
         checked: true,
       })
     ),
   })
 );
 
+/**
+ * Return filtered/highlighted element types
+ * @param {string} searchValue Search term
+ * @return {object} Grouped element types
+ */
+export const getFilteredElementTypes = createSelector(
+  [(state) => state.searchValue],
+  (searchValue) =>
+    highlightMatch(
+      filterNodeGroups(
+        {
+          elementType: Object.entries(sidebarElementTypes).map(([type, name]) => ({
+            id: type,
+            name,
+          })),
+        },
+        searchValue
+      ),
+      searchValue
+    )
+);
+
+/**
+ * Return filtered/highlighted element type items
+ * @param {object} filteredTags List of filtered element types
+ * @param {array} nodeTypes List of node types
+ * @return {object} Element type items
+ */
+export const getFilteredElementTypeItems = createSelector(
```

Comment:
There aren't any related tests set up for this new selector - please add the relevant tests for this under `node-list-items.test.js`